mbedtls/src/hash/mod.rs

```diff
@@ -125,6 +125,36 @@ impl Md {
         }
     }
 
+    pub fn new_hmac(md: Type, key: &[u8]) -> Result<Md> {
```

Comment:
This should be a separate type. It shouldn't be possible to mix hash/hmac functions. The hmac function you added recently should be on the separate type as well.